src/namedtuple/productnamedtuple.jl

```diff
@@ -0,0 +1,155 @@
+"""
+    ProductNamedTupleDistribution{Tnames,Tdists,S<:ValueSupport,eltypes} <:
+        Distribution{NamedTupleVariate{Tnames},S}
+
+A distribution of `NamedTuple`s, constructed from a `NamedTuple` of independent named
+distributions.
+
+Users should use [`product_distribution`](@ref) to construct a product distribution of
+independent distributions instead of constructing a `ProductNamedTupleDistribution`
+directly.
+
+# Examples
+
+```jldoctest ProductNamedTuple; setup = :(using Distributions, Random; Random.seed!(832))
+julia> d = product_distribution((x=Normal(), y=Dirichlet([2, 4])))
+ProductNamedTupleDistribution{(:x, :y)}(
+x: Normal{Float64}(μ=0.0, σ=1.0)
+y: Dirichlet{Int64, Vector{Int64}, Float64}(alpha=[2, 4])
+)
+
+
+julia> nt = rand(d)
+(x = 1.5155385995160346, y = [0.533531876438439, 0.466468123561561])
+
+julia> pdf(d, nt)
```

Comment:
Just today I was dealing with ordering of `NamedTuple` in TransformVariables transforms: https://github.com/tpapp/TransformVariables.jl/pull/120

So I wonder: Should we have
```julia
julia> pdf(d, (; x = 1.5155385995160346, y = [0.533531876438439, 0.466468123561561])) == pdf(d, (; y = [0.533531876438439, 0.466468123561561], x = 1.5155385995160346))
true
```
?

Reply:
Good point, I modified the code to support permutations of the names.